### PR TITLE
Form Validation and Error Handling 

### PR DESCRIPTION
**Pull request by twentymurial33**
Added more validation to make the form functional using "react-hook" ,adding yup and error handling on the form when submitted.

No entries are returned - and it shows 
Screen Shot 2022-11-10 at 2 26 45 PM
Screen Shot 2022-11-10 at 2 26 37 PM

### CODE REVIEW

File: client/src/pages/AddFood.js [hunk 1]
```diff
@@ -42,27 +43,26 @@ function AddFood() {
...
-          <option value="brent">brent</option>
+        <label>Food Type</label>
+        <select {...register("category", { required: true })}>
+          <option value="Breakfast">Breakfast.</option>
+          <option value="Lunch">Lunch</option>
+          <option value="Dinner">Dinner</option>
         </select>
-        <Button variant="contained" onClick={handleSubmit}>
+        <label>Food Description</label>
+        <textarea {...register("aboutYou")} placeholder="About food" />
```
Comment (brentmclark) [inline client/src/pages/AddFood.js:63]: Does `aboutYou` need to be in the schema for this to work as expected?

File: client/src/pages/AddFood.js [hunk 2]
```diff
@@ -42,27 +43,26 @@ function AddFood() {
...
         <h2>Add a New Food Item </h2>
         <label>Food Name</label>
         <input
           type="text"
-          name="food"
-          {...register("food")}
-          onChange={handleChange}
+          {...register("foodName")}
+          placeholder="Food name"
           value={food}
```
Comment (brentmclark) [inline client/src/pages/AddFood.js:53]: Should this `value` be removed now that the field is being managed by `react-hook-form`?

File: client/src/pages/AddFood.js [hunk 3]
```diff
@@ -42,27 +43,26 @@ function AddFood() {
...
-          <p role="alert">Food name is required</p>
-        )}
-        <label>Food Description</label>
-        <textarea></textarea>
-        <label>Food author</label>
-        <select>
-          <option value="murial">murial</option>
-          <option value="brent">brent</option>
+        <label>Food Type</label>
+        <select {...register("category", { required: true })}>
```
Comment (brentmclark) [inline client/src/pages/AddFood.js:57]: Does `category` need to be in the schema for this to work properly?

**Review by brentmclark — COMMENTED** [review]
Looks like this is coming along well!

File: client/src/pages/AddFood.js (hunk 1 above)
Comment (twentymurial33) [reply client/src/pages/AddFood.js]: No it doesn't ..updated it to match what is in the schema file

File: client/src/pages/AddFood.js (hunk 3 above)
Comment (twentymurial33) [reply client/src/pages/AddFood.js]: No it doesn't ..updated it to match what is in the schema file

**Review by brentmclark — APPROVED** [review]
Huzzah!

**Pull request merged**